app/controllers/devise_confirmations_controller.rb

```diff
@@ -22,6 +22,11 @@ def show
     end
   end
 
+  def new
+    super
+    @email = current_user&.unconfirmed_email || current_user&.email
```

Comment:
In the view, we previously had logic that used `pending_reconfirmation?`.  Do we need this here?

Reply:
So i think not here
This is exclusively used to prefill a form to help the user out a bit, the logic should produce:
- current user's unconfirmed email if it exists
- if they don't have one of those then their current confirmed email (in the initial sign up it's current email)
- worst case it's nil, in which case the field value is empty

Reply:
So my hope is the safe navigator and OR syntax give me a concince implimentation for that
(... this is a case where adding sorbet or a ruby explicit type definition might have made my intent clearer. Typed Ruby FTW :trollface:)

Reply:
Ok, that sounds good to me.

Does this view get shown to users who aren't logged in?  If it doesn't, then I can't imagine we need the safe navigation operator on `current_user`.  Also, if that's the case, then maybe we should check that the user actually has an unconfirmed email address (either `unconfirmed_email` on their account, or an `email` that hasn't been confirmed) before rendering the view (e.g. for those who somehow navigate to the page by accident, e.g. using a bookmark).

Reply:
Yeah you can get this from outside a session, so in this case the safe nav operator is that extra guard to ensure we get nil and not error messages

Reply:
Cool, that makes sense 👍